ext/mbstring/tests/sjis2004_encoding.phpt

```diff
@@ -0,0 +1,62 @@
+--TEST--
+Exhaustive test of SJIS-2004 encoding verification and conversion
+--SKIPIF--
+<?php
+extension_loaded('mbstring') or die('skip mbstring not available');
+if (getenv("SKIP_SLOW_TESTS")) die("skip slow test");
+?>
+--FILE--
+<?php
+srand(101); /* Make results consistent */
+include('encoding_tests.inc');
+mb_substitute_character(0x25); // '%'
+
+/* Read in the table of all characters in SJIS-2004 */
+$validChars = array(); /* SJIS-2004 string -> UTF-32BE string */
+$fromUnicode = array(); /* UTF-16BE -> SJIS-2004 */
+$fp = fopen(__DIR__ . '/data/SJIS-2004.txt', 'r+');
+while ($line = fgets($fp, 256)) {
+	if ($line[0] == '#')
+		continue;
+
+	$codepoint2 = null;
+	if (sscanf($line, "0x%x\tU+%x+%x", $bytes, $codepoint1, $codepoint2) >= 2) {
+		$sjis = ($bytes < 256) ? chr($bytes) : pack('n', $bytes);
+		if ($codepoint2) {
+			$validChars[$sjis] = pack('NN', $codepoint1, $codepoint2);
+		} else {
+			/* Two input byte sequences can translate to either a 'halfwidth' or a
+			 * 'fullwidth' version of a character; our implementation of SJIS-2004
+			 * translates them to the fullwidth versions */
+			if (preg_match('/Fullwidth: U\+([0-9A-F]+)/', $line, $match))
```

Comment:
Personally, I like the fact that we use unmodified, "official" conversion tables as the basis for the tests. But if that is what you want, no sweat.

Just say "do it".

Reply:
TBH I'm not sure on this one. Just a bit worried about bloating the distribution with large test-only data files. At least these should gzip well, so maybe I'm worried for nothing.

Reply:
👍🏻 for thinking about the impact on all PHP core developers, not just on `mbstring`.

We could `gzdeflate` the file before parsing it. It would mean that `zlib` has to be enabled to run the `mbstring` tests. It would also mean that if the files are ever modified (which is not likely) we wouldn't get nice diffs.

I believe that `git` stores committed files in a compressed format, so the size of the repo may not increase that much either way. Of course, the size of a checkout will be a bit bigger.

Whichever way you prefer is fine.